File: samples/java/TicTacToe/Server/src/main/java/systems/zlink/samples/tictactoe/server/play/infrastructure/zlink/spots/tictactoegamespot/handlers/TicTacToeGameTimerHandler.java
package systems.zlink.samples.tictactoe.server.play.infrastructure.zlink.spots.tictactoegamespot.handlers;

import systems.zlink.framework.spots.ZLinkSpotTimerHandler;
import systems.zlink.framework.spots.ZLinkTimerTick;
import systems.zlink.samples.tictactoe.server.play.infrastructure.zlink.spots.tictactoegamespot.TicTacToeGame;

import java.util.concurrent.CompletionStage;

// --8<-- [start:doc-timer-handler]
public final class TicTacToeGameTimerHandler implements ZLinkSpotTimerHandler<TicTacToeGame> {
    @Override
    public CompletionStage<Void> handle(TicTacToeGame spot, ZLinkTimerTick tick) {
        return spot.tick();
    }
}
// --8<-- [end:doc-timer-handler]

File: samples/kotlin/TicTacToe/Server/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/server/play/infrastructure/zlink/spots/tictactoegamespot/handlers/TicTacToeGameTimerHandler.kt
package systems.zlink.samples.kotlin.tictactoe.server.play.infrastructure.zlink.spots.tictactoegamespot.handlers

import systems.zlink.framework.kotlin.ZLinkSuspendingSpotTimerHandler
import systems.zlink.framework.spots.ZLinkTimerTick
import systems.zlink.samples.kotlin.tictactoe.server.play.infrastructure.zlink.spots.tictactoegamespot.TicTacToeGame

// --8<-- [start:doc-timer-handler]
class TicTacToeGameTimerHandler() : ZLinkSuspendingSpotTimerHandler<TicTacToeGame> {
    override suspend fun handle(spot: TicTacToeGame, tick: ZLinkTimerTick) = run { spot.tick() }
}
// --8<-- [end:doc-timer-handler]
